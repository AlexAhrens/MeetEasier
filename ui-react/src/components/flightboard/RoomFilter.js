import React, { Component } from 'react';
import PropTypes from 'prop-types';
import * as config from '../../config/flightboard.config.js';

class RoomFilter extends Component {
  constructor(props) {
    super(props);
    // this.state = {
    //   response: false,
    //   error: false,
    //   roomlists: []
    // }
  }

  // getRoomlist = () => {
  //   return fetch('/api/roomlists')
  //     .then((response) => response.json())
  //     .then((data) => {
  //       if(!data.error){
  //         this.setState({
  //           response: true,
  //           error: false,
  //           roomlists: data
  //         });
  //       }
  //       else {
  //         this.setState({
  //           response: true,
  //           error: true,
  //           roomlists: data
  //         });
  //       }
  //     })
  // }

  filterFlightboard = (e) => {
    e.preventDefault();
    this.props.filter(e.target.id);
  }

  componentDidMount() {
    // this.getRoomlist();
  }

  render() {
    const { error, response, roomlists } = this.props;

    return (
      <li>
        <a href="#" className="current-filter">
          {config.roomFilter.filterTitle}
        </a>
        <ul className="menu fb__child-dropdown">
          <li onClick={this.filterFlightboard} id="roomlist-all">
            {config.roomFilter.filterAllTitle}
          </li>

          { response && !error ?
            roomlists.map((item, key) =>
              <li onClick={this.filterFlightboard} id={'roomlist-' + item.toLowerCase().replace(/\s+/g, "-")}>
                {item}
              </li>
            )
          :
            <p>Loading ...</p>
          }

        </ul>
      </li>
    );
  }
}

RoomFilter.propTypes = {
  filter: PropTypes.func
};

export default RoomFilter;
