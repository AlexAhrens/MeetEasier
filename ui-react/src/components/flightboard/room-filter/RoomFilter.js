import React, { Component } from 'react';

let fbConfig = require('../../../config/flightboard.config.js');

class RoomFilter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      response: false,
      roomlists: []
    }
  }

  componentDidMount() {
    return fetch('/api/roomlists')
      .then((response) => response.json())
      .then((data) => {
        this.setState({
          response: true,
          roomlists: data
        });

      })
  }

  filterFlightboard(e) {
    e.preventDefault();
    this.props.filter(e.target.id);
  }

  render() {
    const { response } = this.state;

    return (
      <li>
        <a href="#" className="current-filter">

          {fbConfig.filterTitle}

        </a>
        <ul className="menu fb__child-dropdown">
          <li onClick={this.filterFlightboard.bind(this)} id="roomlist-all">

            {fbConfig.filterAllTitle}

          </li>
          { response ?
            this.state.roomlists.map(function(item, key) {
              return (
                <li onClick={this.filterFlightboard.bind(this)} id={'roomlist-' + item.toLowerCase().replace(/\s+/g, "-")}>{item}</li>
              )
            }.bind(this))
            : <p>Loading ...</p>
          }
        </ul>
      </li>
    );
  }
}

export default RoomFilter;
