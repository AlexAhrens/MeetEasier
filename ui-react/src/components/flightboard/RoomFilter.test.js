import RoomFilter from './RoomFilter';
import fetch from 'isomorphic-fetch';

describe('Flightboard RoomFilter', () => {
  // it('renders correctly', () => {
  //   // const spy = jest.
  //   const wrapper = shallow(<RoomFilter />);
  //
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it('operates filter callback function correctly', () => {
  //   const filterAction = jest.fn();
  //   const wrapper = shallow(<RoomFilter filter={filterAction} />);
  //
  //   wrapper.find('#roomlist-all').simulate('click', {}, 'food');
  //
  //   expect(filterAction).toHaveBeenCalledWith('foo');
  // });
});
