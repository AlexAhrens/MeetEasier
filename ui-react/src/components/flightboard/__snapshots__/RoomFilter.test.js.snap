// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flightboard RoomFilter renders correctly 1`] = `
<li>
  <a
    className="current-filter"
    href="#"
  >
    Locations
  </a>
  <ul
    className="menu fb__child-dropdown"
  >
    <li
      id="roomlist-all"
      onClick={[Function]}
    >
      All Conference Rooms
    </li>
    <p>
      Loading ...
    </p>
  </ul>
</li>
`;
